tests/tabular/checks/model_evaluation/single_dataset_performance_test.py

```diff
@@ -273,3 +273,23 @@ def test_regression_alt_scores_list(diabetes_split_dataset_and_model):
     assert_that(result['max_error'], close_to(-171.719, 0.001))
     assert_that(result['r2'], close_to(0.427, 0.001))
     assert_that(result['neg_mean_absolute_error'], close_to(-45.564, 0.001))
+
+
+def test_non_avg_score_when_more_labels_classes_than_in_data(iris_binary_string_split_dataset_and_model):
+    # Arrange
+    test, train, model = iris_binary_string_split_dataset_and_model
+    check = SingleDatasetPerformance(scorers=['F1 Per Class'])
+    iris_dataset = Dataset(test.data, label='target', label_classes=['a', 'b', 'c', 'dddd'])
```

Comment:
Add an additional test in which the new label is not at the end of the list 
For example: label_classes=['a', 'aa', 'b', 'c']